src/devices/honeywell_cm921.c

```diff
@@ -198,10 +200,19 @@ static data_t *honeywell_cm921_interpret_message(r_device *decoder, const messag
             break;
         }
         case 0x2309: {
-            UNKNOWN_IF(msg->payload_length != 3);
-            data = data_append(data, "zone", "", DATA_INT, msg->payload[0], NULL);
-            // Observation: CM921 reports a very high setpoint during binding (0x7eff); packet: 143255c1230903017efff7
-            data = data_append(data, "setpoint", "", DATA_DOUBLE, ((msg->payload[1] << 8) | msg->payload[2]) * (1 / 100.0F), NULL);
+            UNKNOWN_IF(msg->payload_length % 3 != 0);
```

Comment:
I'm running out of cycles to try to really read this, but it looks like you have figured out a ton!